### PR TITLE
feat(TPG>=5.40.0)!: Add support for RayOperator Addon

### PR DESCRIPTION
**Pull request by genlu2011**
Support for feature was added in 5.40.0 of terraform-provider-google: https://github.com/hashicorp/terraform-provider-google/releases/tag/v5.40.0

container: added field ray_operator_config for resource_container_cluster (https://github.com/hashicorp/terraform-provider-google/pull/18825)

### CODE REVIEW

**Review by apeabody — CHANGES_REQUESTED** [review]
Thanks for the contribution @genlu2011!

It would be best to exercise this new capability in an appropriate existing samples, so it has test coverage: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/tree/master/examples

Even better would to be then validate the cluster state (for Ray Operator Addon) in the respective test verify: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/tree/master/test/integration

File: test/integration/simple_autopilot_public/testdata/TestSimpleAutopilotPublic.json [hunk 1]
```diff
@@ -6,6 +6,15 @@
       "gcePersistentDiskCsiDriverConfig": {
         "enabled": true
       },
+      "rayOperatorConfig": {
+        "enabled": true,
+        "rayClusterLoggingConfig": {
+          "enabled": true
```
Comment (apeabody) [inline test/integration/simple_autopilot_public/testdata/TestSimpleAutopilotPublic.json:12]:
```suggestion
          "enabled": true
```

Comment (genlu2011) [reply test/integration/simple_autopilot_public/testdata/TestSimpleAutopilotPublic.json]: Resolved.

**Comment by genlu2011** [comment]
> Thanks for the contribution @genlu2011!
> 
> It would be best to exercise this new capability in an appropriate existing samples, so it has test coverage: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/tree/master/examples
> 
> Even better would to be then validate the cluster state (for Ray Operator Addon) in the respective test verify: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/tree/master/test/integration

Added test.

**Comment by apeabody** [comment]
Going to re-trigger

```
    private_zonal_with_networking_test.go:79: 
        	Error Trace:	/workspace/test/integration/private_zonal_with_networking/private_zonal_with_networking_test.go:79
        	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.16.1/pkg/tft/terraform.go:638
        	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.16.1/pkg/tft/terraform.go:670
        	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.16.1/pkg/utils/stages.go:31
        	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.16.1/pkg/tft/terraform.go:670
        	Error:      	Should be false
        	Test:       	TestPrivateZonalWithNetworking
        	Messages:   	does not have autoscaling enabled
```

**Comment by apeabody** [comment]
looks like a spacing difference in the test data:

```
Step #94 - "verify simple-autopilot-public-local":         	Error Trace:	/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.16.1/pkg/golden/golden.go:157
Step #94 - "verify simple-autopilot-public-local":         	            				/workspace/test/integration/simple_autopilot_public/simple_autopiliot_public_test.go:59
Step #94 - "verify simple-autopilot-public-local":         	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.16.1/pkg/tft/terraform.go:638
Step #94 - "verify simple-autopilot-public-local":         	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.16.1/pkg/tft/terraform.go:670
Step #94 - "verify simple-autopilot-public-local":         	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.16.1/pkg/utils/stages.go:31
Step #94 - "verify simple-autopilot-public-local":         	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.16.1/pkg/tft/terraform.go:670
Step #94 - "verify simple-autopilot-public-local":         	Error:      	Not equal: 
Step #94 - "verify simple-autopilot-public-local":         	            	expected: "{\n        \"enabled\": true,\n        \"rayClusterLoggingConfig\": {\n          \"enabled\": true\n        },\n        \"rayClusterMonitoringConfig\": {\n          \"enabled\": true\n        }\n      }"
Step #94 - "verify simple-autopilot-public-local":         	            	actual  : "{\n      \"enabled\": true,\n      \"rayClusterLoggingConfig\": {\n        \"enabled\": true\n      },\n      \"rayClusterMonitoringConfig\": {\n        \"enabled\": true\n      }\n    }"
Step #94 - "verify simple-autopilot-public-local":         	            	
Step #94 - "verify simple-autopilot-public-local":         	            	Diff:
Step #94 - "verify simple-autopilot-public-local":         	            	--- Expected
Step #94 - "verify simple-autopilot-public-local":         	            	+++ Actual
Step #94 - "verify simple-autopilot-public-local":         	            	@@ -1,9 +1,9 @@
Step #94 - "verify simple-autopilot-public-local":         	            	 {
Step #94 - "verify simple-autopilot-public-local":         	            	-        "enabled": true,
Step #94 - "verify simple-autopilot-public-local":         	            	-        "rayClusterLoggingConfig": {
Step #94 - "verify simple-autopilot-public-local":         	            	-          "enabled": true
Step #94 - "verify simple-autopilot-public-local":         	            	-        },
Step #94 - "verify simple-autopilot-public-local":         	            	-        "rayClusterMonitoringConfig": {
Step #94 - "verify simple-autopilot-public-local":         	            	-          "enabled": true
Step #94 - "verify simple-autopilot-public-local":         	            	-        }
Step #94 - "verify simple-autopilot-public-local":         	            	+      "enabled": true,
Step #94 - "verify simple-autopilot-public-local":         	            	+      "rayClusterLoggingConfig": {
Step #94 - "verify simple-autopilot-public-local":         	            	+        "enabled": true
Step #94 - "verify simple-autopilot-public-local":         	            	+      },
Step #94 - "verify simple-autopilot-public-local":         	            	+      "rayClusterMonitoringConfig": {
Step #94 - "verify simple-autopilot-public-local":         	            	+        "enabled": true
Step #94 - "verify simple-autopilot-public-local":         	            	       }
Step #94 - "verify simple-autopilot-public-local":         	            	+    }
Step #94 - "verify simple-autopilot-public-local":         	Test:       	TestSimpleAutopilotPublic
Step #94 - "verify simple-autopilot-public-local":         	Messages:   	expected addonsConfig.rayOperatorConfig to match fixture {
Step #94 - "verify simple-autopilot-public-local":         	            	        "enabled": true,
Step #94 - "verify simple-autopilot-public-local":         	            	        "rayClusterLoggingConfig": {
Step #94 - "verify simple-autopilot-public-local":         	            	          "enabled": true
Step #94 - "verify simple-autopilot-public-local":         	            	        },
Step #94 - "verify simple-autopilot-public-local":         	            	        "rayClusterMonitoringConfig": {
Step #94 - "verify simple-autopilot-public-local":         	            	          "enabled": true
Step #94 - "verify simple-autopilot-public-local":         	            	        }
Step #94 - "verify simple-autopilot-public-local":         	            	      }
```

**Comment by genlu2011** [comment]
> looks like a spacing difference in the test data:
> 
> ```
> Step #94 - "verify simple-autopilot-public-local":         	Error Trace:	/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.16.1/pkg/golden/golden.go:157
> Step #94 - "verify simple-autopilot-public-local":         	            				/workspace/test/integration/simple_autopilot_public/simple_autopiliot_public_test.go:59
> Step #94 - "verify simple-autopilot-public-local":         	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.16.1/pkg/tft/terraform.go:638
> Step #94 - "verify simple-autopilot-public-local":         	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.16.1/pkg/tft/terraform.go:670
> Step #94 - "verify simple-autopilot-public-local":         	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.16.1/pkg/utils/stages.go:31
> Step #94 - "verify simple-autopilot-public-local":         	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.16.1/pkg/tft/terraform.go:670
> Step #94 - "verify simple-autopilot-public-local":         	Error:      	Not equal: 
> Step #94 - "verify simple-autopilot-public-local":         	            	expected: "{\n        \"enabled\": true,\n        \"rayClusterLoggingConfig\": {\n          \"enabled\": true\n        },\n        \"rayClusterMonitoringConfig\": {\n          \"enabled\": true\n        }\n      }"
> Step #94 - "verify simple-autopilot-public-local":         	            	actual  : "{\n      \"enabled\": true,\n      \"rayClusterLoggingConfig\": {\n        \"enabled\": true\n      },\n      \"rayClusterMonitoringConfig\": {\n        \"enabled\": true\n      }\n    }"
> Step #94 - "verify simple-autopilot-public-local":         	            	
> Step #94 - "verify simple-autopilot-public-local":         	            	Diff:
> Step #94 - "verify simple-autopilot-public-local":         	            	--- Expected
> Step #94 - "verify simple-autopilot-public-local":         	            	+++ Actual
> Step #94 - "verify simple-autopilot-public-local":         	            	@@ -1,9 +1,9 @@
> Step #94 - "verify simple-autopilot-public-local":         	            	 {
> Step #94 - "verify simple-autopilot-public-local":         	            	-        "enabled": true,
> Step #94 - "verify simple-autopilot-public-local":         	            	-        "rayClusterLoggingConfig": {
> Step #94 - "verify simple-autopilot-public-local":         	            	-          "enabled": true
> Step #94 - "verify simple-autopilot-public-local":         	            	-        },
> Step #94 - "verify simple-autopilot-public-local":         	            	-        "rayClusterMonitoringConfig": {
> Step #94 - "verify simple-autopilot-public-local":         	            	-          "enabled": true
> Step #94 - "verify simple-autopilot-public-local":         	            	-        }
> Step #94 - "verify simple-autopilot-public-local":         	            	+      "enabled": true,
> Step #94 - "verify simple-autopilot-public-local":         	            	+      "rayClusterLoggingConfig": {
> Step #94 - "verify simple-autopilot-public-local":         	            	+        "enabled": true
> Step #94 - "verify simple-autopilot-public-local":         	            	+      },
> Step #94 - "verify simple-autopilot-public-local":         	            	+      "rayClusterMonitoringConfig": {
> Step #94 - "verify simple-autopilot-public-local":         	            	+        "enabled": true
> Step #94 - "verify simple-autopilot-public-local":         	            	       }
> Step #94 - "verify simple-autopilot-public-local":         	            	+    }
> Step #94 - "verify simple-autopilot-public-local":         	Test:       	TestSimpleAutopilotPublic
> Step #94 - "verify simple-autopilot-public-local":         	Messages:   	expected addonsConfig.rayOperatorConfig to match fixture {
> Step #94 - "verify simple-autopilot-public-local":         	            	        "enabled": true,
> Step #94 - "verify simple-autopilot-public-local":         	            	        "rayClusterLoggingConfig": {
> Step #94 - "verify simple-autopilot-public-local":         	            	          "enabled": true
> Step #94 - "verify simple-autopilot-public-local":         	            	        },
> Step #94 - "verify simple-autopilot-public-local":         	            	        "rayClusterMonitoringConfig": {
> Step #94 - "verify simple-autopilot-public-local":         	            	          "enabled": true
> Step #94 - "verify simple-autopilot-public-local":         	            	        }
> Step #94 - "verify simple-autopilot-public-local":         	            	      }
> ```

Yeah, I change the test to compare fields.

**Review by apeabody — APPROVED** [review]

**Pull request merged**